eppo_client/client.py

```diff
@@ -36,143 +34,145 @@ def __init__(
             callback=config_requestor.fetch_and_store_configurations,
         )
         self.__poller.start()
+        self.__evaluator = Evaluator(sharder=MD5Sharder())
 
     def get_string_assignment(
-        self, subject_key: str, flag_key: str, subject_attributes=dict()
+        self,
+        subject_key: str,
+        flag_key: str,
+        subject_attributes: Optional[Dict[str, Union[str, float, int, bool]]] = None,
+        default=None,
     ) -> Optional[str]:
-        try:
-            assigned_variation = self.get_assignment_variation(
-                subject_key, flag_key, subject_attributes, VariationType.STRING
-            )
-            return (
-                assigned_variation.typed_value
-                if assigned_variation is not None
-                else assigned_variation
-            )
-        except ValueError as e:
-            # allow ValueError to bubble up as it is a validation error
-            raise e
-        except Exception as e:
-            if self.__is_graceful_mode:
-                logger.error("[Eppo SDK] Error getting assignment: " + str(e))
-                return None
-            raise e
+        return self.get_assignment_variation(
+            subject_key,
+            flag_key,
+            subject_attributes,
+            VariationType.STRING,
+            default=default,
+        )
+
+    def get_integer_assignment(
+        self,
+        subject_key: str,
+        flag_key: str,
+        subject_attributes: Optional[Dict[str, Union[str, float, int, bool]]] = None,
```

Comment:
I would love to define types or type aliases so we can avoid this but unfortunately that's not supported for Python 3.6

Reply:
What's not supported? Per the [Python docs](https://docs.python.org/3.6/library/typing.html), type aliases were introduced already in 3.5.

I haven't tested it myself, but something like this should work?
```python
AttributeValue = Union[str, float, int, bool]
SubjectAttributes = Dict[str, AttributeValue]
...
subject_attributes: Optional[SubjectAttributes] = None
```